packages/benchmarks/lit-element/list/index.ts

```diff
@@ -214,49 +178,68 @@ export class XApp extends MonitorUpdate {
     container.innerHTML = '';
   };
 
+  const updateComplete = () => new Promise(r => requestAnimationFrame(r));
+
   const benchmark = params.benchmark;
-  const start = 'start';
+  const getTestStartName = (name: string) => `${name}-start`;
+
+  // Named functions are use to run the measurements so that they can be
+  // selected in the DevTools profile flame chart.
 
   // Initial Render
-  let test = 'render';
-  if (benchmark === test || !benchmark) {
-    performance.mark(`${test}${start}`);
-    create();
-    await updateComplete();
-    performance.measure(test, `${test}${start}`);
-    destroy();
+  const render = async () => {
+    const test = 'render';
+    if (benchmark === test || !benchmark) {
+      const start = getTestStartName(test);
+      performance.mark(start);
+      create();
+      await updateComplete();
+      performance.measure(test, start);
+      destroy();
+    }
   }
+  await render();
 
   // Update: toggle data
-  const updateCount = 6;
-  test = 'update';
-  if (benchmark === test || !benchmark) {
-    el = create();
-    performance.mark(`${test}${start}`);
-    for (let i = 0; i < updateCount; i++) {
-      el.items = i % 2 ? otherData : data;
-      await updateComplete();
+  const update = async () => {
+    const updateCount = 6;
+    const test = 'update';
+    if (benchmark === test || !benchmark) {
+      el = create();
+      const start = getTestStartName(test);
+      performance.mark(start);
+      for (let i = 0; i < updateCount; i++) {
+        el.items = i % 2 ? otherData : data;
+        await updateComplete();
```

Comment:
By awaiting rAFs sequentially, this measure _will_ include style/layout/paint. That's ok for some benchmarks, but assuming we're doing the minimal DOM updates, we may want to isolate the LitElement+lit-html work. I think that would require marking the start time at the beginning of the loop body, then adding up the individual times manually.

Reply:
Unfortunately, tachometer doesn't currently support this. I hit this on the repeat benchmark and opened https://github.com/Polymer/tachometer/issues/196.

Seems like we need to decide if the LitElement benchmarks should care about style/layout/paint or just user JS time.  Just measuring JS time seems likely more reliable, so it's a question of what we might miss. Given a certain tree size & choice of styling, do we expect implementation choices to affect style/layout/paint?  If we stay with rAF, we should make sure the time under test is significantly >16ms.